extensions/lifecycle/tasks/LifecycleTask.js

```diff
@@ -1568,48 +1583,50 @@ class LifecycleTask extends BackbeatTask {
         if (obj.VersionId) {
             params.VersionId = obj.VersionId;
         }
-        const req = this.s3target.headObject(params);
-        attachReqUids(req, log);
-        return req.send((err, data) => {
-            LifecycleMetrics.onS3Request(log, 'headObject', 'bucket', err);
-
-            if (err) {
+        const command = new HeadObjectCommand(params);
+        attachReqUids(command, log);
+        return this.s3target.send(command)
+            .then(data => {
+                LifecycleMetrics.onS3Request(log, 'headObject', 'bucket', null);
+                const lastModified = data.LastModified instanceof Date 
+                    ? data.LastModified.toISOString() 
+                    : data.LastModified;
```

Comment:
we could do this to cover all cases:
```suggestion
                const lastModified = new Date(data.LastModified).toISOString();
```
...or do you think there is a performance penalty/risk ?

It seems strange we can receive either a Date or "something" else (a string?) : did you see it, or is it some leftover from troubleshooting? Or maybe it handles some null/undefined case, which could & should be handled in a more explicit way?

Reply:
I double check this, headObject returns either a Date or undefined. But actually, it was the same in sdk v2, which means this change should be useless as the behavior shouldn't be modified. Yet I had some issues around the lastModified date in the ci pipeline so its weird. 
I'm gonna remove it for now, I think it's useless, but later I will rerun the pipeline to see if the error comes back

Reply:
And yeah so I looked more, and the real important modification happens in HeadObjectCommand, in lifecycleDeleteObjectTask.js line ~100 : 
There is a IfUnmodifiedSince parameter that used to take lastModified as a parameter,
In sdk v2, this parameter was Date | undefined
But in sdkv3, it cannot be undefined, must be a Date.